test/test_type.jl

```diff
@@ -686,6 +686,75 @@ isdir(outdir) && rm(outdir, recursive = true)
         end
     end
 
+    @timed_testset "Compressible Navier Stokes Diffusion 1D" begin
+        for RealT in (Float32, Float64)
+            equations = @inferred CompressibleEulerEquations1D(RealT(1.4))
+            prandtl_number() = RealT(0.72)
+            mu() = RealT(0.01)
+            equations_parabolic_primitive = @inferred CompressibleNavierStokesDiffusion1D(equations,
+                                                                                          mu = mu(),
+                                                                                          Prandtl = prandtl_number(),
+                                                                                          gradient_variables = GradientVariablesPrimitive())
+            equations_parabolic_entropy = @inferred CompressibleNavierStokesDiffusion1D(equations,
+                                                                                        mu = mu(),
+                                                                                        Prandtl = prandtl_number(),
+                                                                                        gradient_variables = GradientVariablesEntropy())
+
+            x = SVector(zero(RealT))
+            t = zero(RealT)
+            u = u_inner = u_transformed = flux_inner = SVector(one(RealT), zero(RealT),
+                                                               zero(RealT))
+            orientation = 1
+            directions = [1, 2]
+            gradients = SVector(RealT(0.1), RealT(0.1), RealT(0.1), RealT(0.1))
+
+            for equations_parabolic in (equations_parabolic_primitive,
+                                        equations_parabolic_entropy)
+                @test eltype(@inferred flux(u, gradients, orientation, equations_parabolic)) ==
+                      RealT
+
+                @test eltype(@inferred cons2prim(u, equations_parabolic)) == RealT
+                @test eltype(@inferred prim2cons(u, equations_parabolic)) == RealT
+                @test eltype(@inferred cons2entropy(u, equations_parabolic)) == RealT
+                @test eltype(@inferred entropy2cons(u, equations_parabolic)) == RealT
+                @test typeof(@inferred Trixi.temperature(u, equations_parabolic)) == RealT
+
+                @test eltype(@inferred Trixi.convert_transformed_to_primitive(u_transformed,
+                                                                              equations_parabolic)) ==
+                      RealT
+                @test eltype(@inferred Trixi.convert_derivative_to_primitive(u, gradients,
+                                                                             equations_parabolic)) ==
+                      RealT
+            end
+
+            # test for GradientVariablesPrimitive
```

Comment:
Tests here are really tedious but these functions are not modified actually. Do you think it is necessary to test these functions? @ranocha 

If you think it is necessary, I will do it later.

Reply:
You could also open an issue and list which kind of tests are missing. Then, they could be added later (also by other people).